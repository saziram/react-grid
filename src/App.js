import './App.css';
import { times } from 'lodash';
import React, { useState } from 'react';

function App() {
  
  const rows = 5;
  const columns = 6; 
  const [isClicked, setIsClicked] = useState(false);
  const [coords, setCoords] = useState({
    startAt: '',
    endAt: ''
  }); 
  const onClickHandle = e => { 
    setIsClicked(e.type === 'pointerdown');
    if(e.type === 'pointerdown'){
      setCoords({...coords, startAt: e.target.id});
    }else{
      clearStyles();
    }

  };

  const clearStyles = () => {
    const box = document.getElementsByClassName("box");
    [].forEach.call(box, (element) => {
      element.style.backgroundColor = '#fff';
    }); 
  }

  const onPointerMoveHandle = e => {
    if(isClicked){
      setCoords({...coords, endAt: e.target.id});
      const startAt = coords.startAt.split("-");
      const endAt = coords.endAt.split("-");
      clearStyles();
      startAt.forEach((element, index) => {
        if(element > endAt[index]){
          startAt[index] = endAt[index];
          endAt[index] = element;
        }
      });
      for(let i = startAt[0]; i <= endAt[0]; i++){
        for(let j = startAt[1]; j <= endAt[1]; j++){
          document.getElementById(`${i}-${j}`).style.backgroundColor = '#ddd';
        }   
      }      
    }
  };

  return (
    <div className="App">
      <div id="root" className="root">
      {
        times( rows, r => <div key={r} className="row">
          { times( columns, c => <div key={`${r}-${c}`} id={`${r+1}-${c+1}`} className="box" onPointerDown={onClickHandle} onPointerUp={onClickHandle} onPointerEnter={onPointerMoveHandle} ></div>) }
        </div>  
        ) 
      }
      </div>    
    </div>
  );
}

export default App;
