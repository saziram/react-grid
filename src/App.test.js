import { render, screen } from '@testing-library/react';
import App from './App';
import {create} from 'react-test-renderer'; // ES6

describe('Testing Grid', () => {

  const rows = 5;
  const columns = 6; 

  const component = create( <App /> );
  const element = component.root.findByType("div");


  test('should matches the snapshot', () => {    
    expect(component.toJSON()).toMatchSnapshot();
  });



});
