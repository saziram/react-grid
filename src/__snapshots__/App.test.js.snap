// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Grid should Matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="root"
    id="root"
  >
    <div
      className="row"
    >
      <div
        className="box"
        id="1-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="2-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="3-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="4-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="5-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Testing Grid should matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="root"
    id="root"
  >
    <div
      className="row"
    >
      <div
        className="box"
        id="1-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="2-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="3-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="4-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="5-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Testing Grid should render grid 1`] = `
<div
  className="App"
>
  <div
    className="root"
    id="root"
  >
    <div
      className="row"
    >
      <div
        className="box"
        id="1-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="1-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="2-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="2-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="3-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="3-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="4-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="4-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="box"
        id="5-1"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-2"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-3"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-4"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-5"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
      <div
        className="box"
        id="5-6"
        onPointerDown={[Function]}
        onPointerEnter={[Function]}
        onPointerUp={[Function]}
      />
    </div>
  </div>
</div>
`;
